frame/contracts/proc-macro/Cargo.toml

```diff
@@ -17,7 +17,7 @@ proc-macro = true
 [dependencies]
 proc-macro2 = "1"
 quote = "1"
-syn = "1.0.98"
+syn = { version = "1.0.98", features = ["full"] }
```

Comment:
Fix from https://github.com/paritytech/substrate/pull/12706 so that `cargo check -p pallet-contracts-proc-macro` passes in this PR